manual/developer/admin/README.md

```diff
@@ -0,0 +1,323 @@
+and thanks to debouncing this should not affect performance. The schema refresh process uses a few
+auxiliary components that may have different implementations depending on the Cassandra version:
+
+* `SchemaQueries`: launches the schema queries asynchronously, and assemble the result in a
+  `SchemaRows`;
+* `SchemaParser`: turns the `SchemaRows` into the `SchemaRefresh`. 
+
+When the metadata manager needs node-related data, it queries the topology monitor. When it needs
+schema-related data, it uses the control connection directly to issue its queries.
+
+### Topology monitor
+
+`TopologyMonitor` abstracts how we get information about nodes in the cluster:
+
+* refresh the list of nodes;
+* refresh an individual node, or load the information of a newly added node;
+* check schema agreement;
+* emit `TopologyEvent` instances on the bus when we get external signals suggesting topology changes
+  (node added or removed), or status changes (node down or up).
+
+The built-in implementation uses the control connection to query `system.local` and `system.peers`,
+and listen to gossip events.
+
+### Node state manager
+
+`NodeStateManager` tracks the state of the nodes in the cluster.
+
+We can't simply trust gossip events because they are not always reliable (the coordinator can become
+isolated and think other nodes are down). Instead, the driver uses more elaborate rules that combine
+external signals with observed internal state:
+
+* as long as we have an active connection to a node, it is considered up, whatever gossip events
+  say;
+* if all connections to a node are lost, and its pool has started reconnecting, it gets marked down
+  (we check the reconnection because the pool could have shut down for legitimate reasons, like the
+  node distance changing to IGNORED);
+* a node is marked back up when the driver has successfully reopened at least one connection;
+* if the driver is not actively trying to connect to a node (for example if it is at distance
+  IGNORED), then gossip events are applied directly.
+  
+See the javadocs of `NodeState` and `TopologyEvent`, as well as the `NodeStateManager`
+implementation itself, for more details.
+
+#### Topology events vs. node state events
+
+These two event types are related, but they're used at different stages:
+
+* `TopologyEvent` is an external signal about the state of a node (by default, a `TOPOLOGY_CHANGE`
+  or `STATUS_CHANGE` gossip event received on the control connection). This is considered as a mere
+  suggestion, that the driver may or may not decide to follow;
+* `NodeStateEvent` is an actual decision made by the driver to change a node to a given state.
+
+`NodeStateManager` essentially transforms topology events, as well as other internal signals, into
+node state events.
+
+In general, other driver components only react to node state events, but there are a few exceptions:
+for example, if a connection pool is reconnecting and the next attempt is scheduled in 5 minutes,
```

Comment:
I wonder if we should add the link to the actual code where this is happening?
Or we don't want to expose links to internal code because of the maintenance burden that it could cause?